crypto/bio/bss_dgram.c

```diff
@@ -1116,17 +1117,19 @@ static int dgram_sctp_read(BIO *b, char *out, int outl)
 #  endif
                 }
 #  ifdef SCTP_AUTHENTICATION_EVENT
-                if (snp->sn_header.sn_type == SCTP_AUTHENTICATION_EVENT)
-                    dgram_sctp_handle_auth_free_key_event(b, snp);
+                if (snp.sn_header.sn_type == SCTP_AUTHENTICATION_EVENT)
+                    dgram_sctp_handle_auth_free_key_event(b, &snp);
 #  endif
 
                 if (data->handle_notifications != NULL)
                     data->handle_notifications(b, data->notification_context,
                                                (void *)out);
 
+                OPENSSL_cleanse(&snp, sizeof(snp));
                 memset(out, 0, outl);
```

Comment:
It seems strange to "cleanse" `snp` but to memset `out`. They're either both sensitive (in which case "cleanse" is appropriate" for both), or they're not, in which case memset is sufficient for both (or in fact not necessary at all...)

Reply:
In case they are sensitive, snp has to be cleansed because its a local stack variable. In case of out it depends on the context in which the function is called. If the buffer it points to is always application provided, compiler would not be allowed to optimize the memset.

In case they are not sensitive, you would not even have to memset snp.

So in case of snp its either cleanse or nothing. Not knowing the code here well enough here, i went with the conservative choice. Maybe someone who is more into sctp can comment.

Reply:
I find it difficult to believe that this structure holds sensitive data. @tuexen might have an opinion.

Reply:
Not sure what sensitive data means... It is metadata from the SCTP stack. Something like a path state has changed (so a remote IP address, a state), key management information (a key identifier), or similar information.

Reply:
Sensitive means e.g., key material. Id consder the information you mentioned non-sensitive. Replaced the cleanse by a normal memset.

Reply:
There is no key material involved.